contrib/platform/src/com/sun/jna/platform/win32/Crypt32.java

```diff
@@ -142,4 +145,203 @@ public boolean CryptUnprotectData(DATA_BLOB pDataIn, PointerByReference szDataDe
+	 *         function fails, it returns zero (FALSE).
+	 */
+	boolean CertGetCertificateChain(Pointer hChainEngine, CERT_CONTEXT pCertContext, Pointer pTime,
+			Pointer hAdditionalStore, CERT_CHAIN_PARA pChainPara, int dwFlags, Pointer pvReserved,
+			CERT_CHAIN_CONTEXT ppChainContext);
+
+	/**
+	 * The CertFreeCertificateContext function frees a certificate context by
+	 * decrementing its reference count. When the reference count goes to zero,
+	 * CertFreeCertificateContext frees the memory used by a certificate context.
+	 *
+	 * @param pCertContext
+	 *            A pointer to the CERT_CONTEXT to be freed.
+	 * @return The function always returns nonzero.
+	 */
+	boolean CertFreeCertificateContext(CERT_CONTEXT pCertContext);
+
+	/**
+	 * The CertCloseStore function closes a certificate store handle and reduces the
+	 * reference count on the store. There needs to be a corresponding call to
+	 * CertCloseStore for each successful call to the CertOpenStore or
+	 * CertDuplicateStore functions.
+	 *
+	 * @param hCertStore
+	 *            Handle of the certificate store to be closed.
+	 * @param dwFlags
+	 *            Typically, this parameter uses the default value zero. The default
+	 *            is to close the store with memory remaining allocated for contexts
+	 *            that have not been freed. In this case, no check is made to
+	 *            determine whether memory for contexts remains allocated.
+	 * @return If the function succeeds, the return value is TRUE. If the function
+	 *         fails, the return value is FALSE.
+	 */
+	boolean CertCloseStore(HANDLE hCertStore, int dwFlags);
+
+	/**
+	 * The CertNameToStr function converts an encoded name in a CERT_NAME_BLOB
+	 * structure to a character string.
+	 *
+	 * @param dwCertEncodingType
+	 *            The certificate encoding type that was used to encode the name.
+	 *            The message encoding type identifier, contained in the high WORD
+	 *            of this value, is ignored by this function.
+	 * @param pName
+	 *            A pointer to the CERT_NAME_BLOB structure to be converted.
+	 * @param dwStrType
+	 *            This parameter specifies the format of the output string. This
+	 *            parameter also specifies other options for the contents of the
+	 *            string.
+	 * @param psz
+	 *            A pointer to a character buffer that receives the returned string.
+	 *            The size of this buffer is specified in the csz parameter.
+	 * @param csz
+	 *            The size, in characters, of the psz buffer. The size must include
+	 *            the terminating null character.
+	 * @return Returns the number of characters converted, including the terminating
+	 *         null character. If psz is NULL or csz is zero, returns the required
+	 *         size of the destination string.
+	 */
+	int CertNameToStr(int dwCertEncodingType, Pointer pName, int dwStrType, char[] psz, int csz);
```

Comment:
`pName` is a `PCERT_NAME_BLOB` which is typedeffed from `DATA_BLOB`. Can the correct type be used here?

The `char[]` mapping for `psz` is problematic. This is correct if the system property `w32.ascii` is not set. This property determines if the Ascii or the unicode variant of the type- and functionmapper is used by default.

In this case I'd advise this: Bind `CertNameToStr`  with Pointer as `psz` parameter and add utility method to Crypt32Util. That function would determine the needed size of the memory block to hold the name (depending on the value of the mentioned system property the factor 2 (unicode in windows is UCS2), allocate that memory, call the function and read the string from the memory. 

See the comment about the return value:

> Returns the number of characters converted, including the terminating null character.
> If psz is NULL or csz is zero, returns the required size of the destination string.